cfc-testing/scripts/comprehensive-synth-test.sh

```
@@ -23,6 +31,46 @@ CDK_OUT_DIR="$BASE_DIR/cdk.out"
 
+# Function to parse cdk-nag CSV reports for warnings (L1 advisories)
+# Returns count of warnings found and prints them
+parse_cdknag_csv_warnings() {
+    local stack_name="$1"
+    local warning_count=0
+
+    # Find cdk-nag CSV reports for this stack
+    for csv_file in "$CDK_OUT_DIR"/*-"${stack_name}"-NagReport.csv "$CDK_OUT_DIR"/*NagReport.csv; do
+        if [ -f "$csv_file" ]; then
+            # Extract pack name from filename (e.g., "AwsSolutions" from "AwsSolutions-stack-NagReport.csv")
+            local pack_name=$(basename "$csv_file" | cut -d'-' -f1)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)